sirepo/pkcli/cloudmc.py

```diff
@@ -74,29 +61,34 @@
 
 
 def extract_dagmc(dagmc_filename):
-    mat = _extract_volumes(dagmc_filename)
-    for vol in mat.values():
-        _vtk_to_bin(vol.volId)
-    return mat
+    mb = pymoab.core.Core()
+    mb.load_file(dagmc_filename)
+    visited = set()
+    res = PKDict()
+    work_items = []
+    for name, volumes in _groups_and_volumes(mb).items():
+        if _visited_volume(volumes, visited):
+            continue
+        vol_id = _get_tag_value(
+            mb,
+            mb.tag_get_handle(pymoab.types.GLOBAL_ID_TAG_NAME),
+            volumes[0],
+        )
+        res[name] = PKDict(
+            name=name,
+            volId=vol_id,
+        )
+        work_items.append((dagmc_filename, vol_id, volumes))
+    work_items.sort(key=lambda v: -len(v[2]))
```

Comment:
The chunk size should have been left off - the default is size 1, I think. I moved the mutiprocessing logic into sirepo.mpi, and it could be replaced with threading if that makes more sense.